src/vendors/VendorDetails.tsx

```diff
@@ -47,8 +58,12 @@ const VendorDetails = ({ vendor, close, action }: Props) => {
               <Phone />
             </TouchableOpacity>
           )}
-          {location && (
-            <TouchableOpacity onPress={() => initiateDirection({ coordinate: location })}>
+          {((location.latitude !== 0 && location.longitude !== 0) || street) && (
+            <TouchableOpacity
+              onPress={() =>
+                initiateDirection({ title, coordinate: location, building_number, street, city })
+              }
```

Comment:
Should be wrapped in a function, since `initiateDirection` can throw an error. The function should catch the error and use `dispatch(showError(msg:string))` so that it surfaces an error message to the user.